core/style.js

```diff
@@ -2117,3 +2117,81 @@ CKEDITOR.tools.extend( CKEDITOR.editor.prototype, {
  * @cfg {String/Array/Boolean} [stylesSet='default']
  * @member CKEDITOR.config
  */
+
+/**
+ * Abstract class describing definition of a style.
+ * It is used as one of {@link CKEDITOR.style#constructor}'s arguments.
+ *
+ * Style definition object represents a style as a set of properties describing CSS style rules and HTML attributes.
+ * The {@link CKEDITOR.style} based on such definition can be applied to and removed from selection
+ * through various {@link CKEDITOR.style} methods.
+ *
+ *  	{
+ *			name: 'Special Title',
+ *			element: 'h1',
+ *			attributes: { class: 'my_class' },
+ *			styles: { color: 'red', 'font-size': '16px', 'font-width': 'bold' }
+ *		}
+ *
+ *	More on how editor content styles are handled can be found in the [Styles guide](#!/guide/dev_howtos_styles).
+ *
+ * @class CKEDITOR.style.definition
+ * @abstract
+ */
+
+/**
+ * Defines style type.
+ *
+ * There are three standard style types - {@link CKEDITOR#STYLE_INLINE}, {@link CKEDITOR#STYLE_BLOCK}
+ * or {@link CKEDITOR#STYLE_OBJECT}, each one related to the element used in the style rule andq types of
+ * elements to which specific style can be applied.
+ *
+ * Plugins may define {@link CKEDITOR.style.customHandlers special style handlers} which can be applied in certain situations.
+ * To use special style handler, `type` property should be set as a name of the style handler, e.g. `widget`.
+ *
+ * More on style types can be found in the [Style Types section of the Styles guide](#!/guide/dev_styles-section-style-types).
+ *
+ * @property {String/Number} type=CKEDITOR.STYLE_INLINE
+ */
+
+/**
+ * An unique name assigned to a style definition.
+ *
+ *		{ name: 'Special title' }
+ *
+ * @property {String} name
+ */
+
+/**
+ * A set of properties specifying attributes of the HTML style element.
+ * If `style` attribute is present it will be merged with existing {@link CKEDITOR.style.definition#styles} property.
+ *
+ *		{
+ *			attributes: {
+ *				style: 'color: red',
+ *				class: 'link'
+ *			}
+ *		}
+ *
+ * @property {Object} attributes
+ */
+
+/**
+ * Element type which will be applied to a selection when applying a style. It should be a valid HTML element, for example `span`.
+ *
+ *		{ element: 'h1' }
+ *
+ * @property {String} element
+ */
+
+/**
+ * A set of properties specifying CSS style rules.
+ *
+ *		{
```

Comment:
It's inconsistent with code listings in `attributes` and `element`. You should wrap it with another object and add `styles` property, and then there should be an object that we see here.